### DISCUSSION TITLE
Add selfhostable search engine

### DISCUSSION THREAD

**Discussion by lottotto** [google/docsy#1989] [Ideas]
Hello
I am using the Docsy theme to develop internal product documentation for my enterprise. As our documentation grows, Lunr's client-side search functionality is reaching its limits. Therefore, it would be beneficial to integrate with self-hosted search solutions such as Typesense, Meilisearch, and Elasticsearch.

Personally, I find Meilisearch's user interface appealing and would prefer to implement it for search purposes. I would greatly appreciate any assistance you can provide. What are your thoughts on this matter?

Thanks

**Comment by fekete-robert** [comment]
I guess that highly depends on the search service that you use, and how that can ingest data (does it crawl links to update its index, or does it expect a specific format like JSON to ingest data, and so on).

For accessing the search page, I think the easiest is to add a search entry to the menu that links to your search page - at least at first I wouldn't try to integrate it into the docs.

**Comment by lottotto** [comment]
got it. thank you.

I intended to display search results. I think it's easy to add it to the search screen.
I'll try implementing it myself.
thank you.
